File: examples/quickstart.py
"""Search for a company and print its disclosure history.

Usage: MYCELIUM_API_KEY=... python quickstart.py "unilever"
"""

import os
import sys

import requests

BASE = "https://api.mycelium.global/v1"
HEADERS = {"X-Api-Key": os.environ["MYCELIUM_API_KEY"]}


def main(name: str) -> None:
    search = requests.get(
        f"{BASE}/entities/search", params={"name": name}, headers=HEADERS, timeout=30
    ).json()
    if not search["results"]:
        print(f"No entities matched {name!r}")
        return

    entity = search["results"][0]
    public_id = entity["publicId"]
    print(f"{entity['legalName']} (publicId {public_id}, Mycelium Score {entity['myceliumScore']})")

    disclosures = requests.get(
        f"{BASE}/entities/{public_id}/disclosures", headers=HEADERS, timeout=30
    ).json()
    for disclosure in disclosures["results"]:
        total = disclosure["total"]["value"] if disclosure.get("total") else None
        print(f"  {disclosure['year']}: total {total} kgCO2e (disclosure {disclosure['publicId']})")


if __name__ == "__main__":
    main(sys.argv[1] if len(sys.argv) > 1 else "unilever")

File: examples/quickstart.js
// Search for a company and print its disclosure history.
// Usage: MYCELIUM_API_KEY=... node quickstart.js "unilever"

const BASE = "https://api.mycelium.global/v1";
const headers = { "X-Api-Key": process.env.MYCELIUM_API_KEY };

async function main(name) {
  const search = await fetch(
    `${BASE}/entities/search?name=${encodeURIComponent(name)}`,
    { headers },
  ).then((r) => r.json());

  if (!search.results.length) {
    console.log(`No entities matched "${name}"`);
    return;
  }

  const entity = search.results[0];
  console.log(
    `${entity.legalName} (publicId ${entity.publicId}, Mycelium Score ${entity.myceliumScore})`,
  );

  const disclosures = await fetch(
    `${BASE}/entities/${entity.publicId}/disclosures`,
    { headers },
  ).then((r) => r.json());

  for (const disclosure of disclosures.results) {
    const total = disclosure.total ? disclosure.total.value : null;
    console.log(
      `  ${disclosure.year}: total ${total} kgCO2e (disclosure ${disclosure.publicId})`,
    );
  }
}

main(process.argv[2] ?? "unilever");
